ts/run.tsx

```diff
@@ -79,15 +79,15 @@ class Viewer {
 				.pipe(tracker) // mutators -> latest gamestate
 				.pipe(scrubber) // gamestate -> gamestate emit on scrub past
 				.pipe(sink); // gamestate
-			if(preloader) {
+			if(preloader.canPreload()) {
```

Comment:
Can you please change this to `if(this.opts.textureDirectory) {`? Otherwise it makes no sense to pipe the mutators for texture extraction into the preloader.